src/lib.rs

```diff
@@ -479,12 +479,20 @@ macro_rules! impl_numtoa_base_n_init_for {
     $base:expr,
     $core_function_name:ident,
     $base_n_function_name:ident,
+    $padded_function_name:ident,
     $needed_buffer_size:expr) => {
 
         pub const fn $base_n_function_name(num: $type_name) -> AsciiNumber<$needed_buffer_size> {
             let mut string = [0_u8; $needed_buffer_size];
-            let len = $core_function_name(num, $base, &mut string).len();
-            return AsciiNumber { string, start:$needed_buffer_size-len}
+            let start = $needed_buffer_size - $core_function_name(num, $base, &mut string).len();
+            return AsciiNumber { string, start }
+        }
+
+        pub const fn $padded_function_name<const LENGTH: usize>(num: $type_name, padding: u8) -> AsciiNumber<LENGTH> {
+            const { assert!(LENGTH >= $needed_buffer_size) }
```

Comment:
static assertion that the requested length with padding is always at least the length of the string being generated